src/rdkafka_range_assignor.c

```diff
@@ -55,12 +55,11 @@ rd_kafka_resp_err_t rd_kafka_range_assignor_assign_cb(
     const rd_kafka_assignor_t *rkas,
     const char *member_id,
     const rd_kafka_metadata_t *metadata,
+    const rd_kafka_metadata_internal_t *metadata_internal,
```

Comment:
Question for here and elsewhere, why pass both metadata and metadata_internal? Doesn't it suffice to pass just metadata_internal, as it contains a reference to metadata?

Also, another question: wouldn't it make sense (from a pointer casting point of view) for partition_internal to have a reference to partition, and broker_internal to have a reference to broker (rather than the Ids),The sorting, if required, could still work based on the Ids, since they are still accessible.

If we do that, we only need to pass _internal fields around in librdkafka private code. 
One drawback may be, this change will become larger since everywhere we will need an extra "->".

And same for topic_internal and topic - though I notice that internal_topic does not have any way to map it to topic yet.

Reply:
I was thinking to change this, also because there are requests for a custom assignor and we can't pass rd_kafka_metadata_internal_t there. Instead it could use accessor functions to get required data.

Reply:
Oh, right, that would make it way more difficult anyway. If we follow this comment and mark it as experimental: https://github.com/confluentinc/librdkafka/issues/2284#issuecomment-774961322
But then, will you have a typedef for rd_kafka_metadata_internal_t in rdkafka.h along with all accessor methods, and treat it like any other private struct?

Reply:
We keep `rd_kafka_metadata_internal_t `, we only have accessor methods in `rdkafka.h`. For rack_id for example.